File: vulnapp.java
import java.io.*;
import java.sql.*;

public class VulnerableApp {
    
    // Hardcoded credentials (Bad Practice)
    private static final String USERNAME = "admin";
    private static final String PASSWORD = "password123";

    public static void main(String[] args) throws Exception {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        
        System.out.print("Enter username: ");
        String userInput = reader.readLine();
        System.out.print("Enter password: ");
        String passInput = reader.readLine();
        authenticate(userInput, passInput);
        
        System.out.print("Enter SQL query: ");
        String sqlQuery = reader.readLine();
        sqlInjectionVulnerable(sqlQuery);
        
        System.out.print("Enter command: ");
        String cmd = reader.readLine();
        commandInjectionVulnerable(cmd);
        
        System.out.print("Enter serialized object file: ");
        String fileName = reader.readLine();
        insecureDeserialization(fileName);
    }
    
    private static void authenticate(String user, String password) {
        if (USERNAME.equals(user) && PASSWORD.equals(password)) {
            System.out.println("Authentication successful!");
        } else {
            System.out.println("Authentication failed!");
        }
    }
    
    private static void sqlInjectionVulnerable(String query) {
        try (Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/testdb", "root", "root");
             Statement stmt = conn.createStatement()) {
            
            // Vulnerable SQL query (No prepared statement, direct concatenation)
            ResultSet rs = stmt.executeQuery("SELECT * FROM users WHERE name = '" + query + "'");
            while (rs.next()) {
                System.out.println("User: " + rs.getString("name"));
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
    
    private static void commandInjectionVulnerable(String command) {
        try {
            // Directly executing user input (Vulnerable to command injection)
            Runtime.getRuntime().exec(command);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    
    private static void insecureDeserialization(String fileName) {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(fileName))) {
            // Insecure deserialization of untrusted data
            Object obj = ois.readObject();
            System.out.println("Deserialized object: " + obj.toString());
        } catch (IOException | ClassNotFoundException e) {
            e.printStackTrace();
        }
    }
}

File: vulnapp.py
import sqlite3
import os
import pickle

# Hardcoded credentials (Bad Practice)
USERNAME = "admin"
PASSWORD = "password123"

def authenticate(user, password):
    if user == USERNAME and password == PASSWORD:
        print("Authentication successful!")
    else:
        print("Authentication failed!")

def sql_injection_vulnerable(query):
    conn = sqlite3.connect(":memory:")
    cursor = conn.cursor()
    cursor.execute("CREATE TABLE users (id INTEGER PRIMARY KEY, name TEXT, password TEXT)")
    cursor.execute("INSERT INTO users (name, password) VALUES ('admin', 'adminpass')")
    
    # Vulnerable SQL query (no parameterized query, direct concatenation)
    cursor.execute("SELECT * FROM users WHERE name = '" + query + "'")
    print(cursor.fetchall())
    conn.close()

def command_injection_vulnerable(command):
    # Dangerous: Directly executing user input in system command
    os.system(command)

def insecure_deserialization(serialized_data):
    # Unsafe deserialization of untrusted data
    return pickle.loads(serialized_data)

def eval_execution(expression):
    # Arbitrary code execution vulnerability
    return eval(expression)

if __name__ == "__main__":
    user_input = input("Enter username: ")
    pass_input = input("Enter password: ")
    authenticate(user_input, pass_input)
    
    sql_query = input("Enter SQL query: ")
    sql_injection_vulnerable(sql_query)
    
    cmd = input("Enter command: ")
    command_injection_vulnerable(cmd)
    
    pickled_data = input("Enter pickled data: ").encode()
    insecure_deserialization(pickled_data)
    
    expr = input("Enter Python expression: ")
    print(eval_execution(expr))
